### DISCUSSION TITLE
Apps are blank boxes

### DISCUSSION THREAD

**Discussion by ghost** [whs/runekit#21] [Q&A Linux]
Hi, I appreciate this project and was curious if I could get it running but have run into some issues. All the apps are just blank boxes with no content. Using the Linux AppImage (chmod +x)

With 'allow applications to block compositing' I get: 
./RuneKit.AppImage 
INFO:root:Starting QtWebEngine
INFO:runekit.game.x11.manager.X11GameManager:Found game instance 60817419
WARNING:runekit.game.overlay.DesktopWideOverlay:Detected black screen condition. Disabling overlay

Without 'allow applications to block compositing' I don't get the disabled overlay but all the apps are still blank. 
DE: Plasma 5.24.1
WM: Kwin

I have tried other desktop environments and window managers with no luck. Also tried package manager and flatpak versions of the client. RuneKit is in the tray but any app selected is empty. Any tips? Thanks!

**Comment by willowew** [comment]
I also have this problem. It seems that this could have been resolved previously by changing the rendering backend to Xrender, KDE did unfortunately removed this functionality not so long ago. I have gotten it to work with Linux Mint together with the MATE DE, using the provided Appimage of Runekit and flatpak version of RS3. If you happen to find a solution and make it work with Plasma, please let me know!

**Reply by Pen-thus** [reply]
I encountered this on Manjaro + xfce/xfwm4 today after updating stable y'day (and again today) - doing just `./RuneKit.AppImage` results in the runekit icon is added to tray, but trying to launch anything results in blank apps
However, on the basis of [this thread](https://github.com/vlaci/openconnect-sso/issues/69), doing `export QTWEBENGINE_CHROMIUM_FLAGS="--no-sandbox"` in terminal before `./RuneKit.AppImage` does the trick and I can open my plugins again
So, putting it all together, doing `export QTWEBENGINE_CHROMIUM_FLAGS="--no-sandbox" & ./RuneKit.AppImage` should do the trick
It seems to be a recent delta in some qt functionality, but I have no comprehension of programming so I can't say how or why this occurred
I hope this helps you

**Reply by willowew** [reply]
Great, that worked. It fixed the blank screen issue that appeared when one app in Runekit is selected. Unfortunetly there seems to be another problem related to this. After launching an app such as afkwarden from the tray (with Runekit opened from the terminal with the aforementioned solution), it opens as it's expected to. However, when pressing any button inside of afkwarden, to configure it for example, it now seems to be having the same black screen. It seems to be the same for all apps that open a new window when settings are adjusted (such as afkwarden). This is what appears in the terminal:

export QTWEBENGINE_CHROMIUM_FLAGS="--no-sandbox" & ./RuneKit.AppImage

INFO:root:Starting QtWebEngine  
INFO:runekit.game.x11.manager.X11GameManager:Found game instance 65011723  
WARNING:runekit.game.overlay.DesktopWideOverlay:Detected black screen condition. Disabling overlay  
QQuickWidget: Failed to make context current when creating FBO  
qt.qpa.backingstore: composeAndFlush: makeCurrent() failed

Any idea on how to fix this? Really appreciate the help!

**Reply by Pen-thus** [reply]
Given that there's an identifiable difference between the current runekit configuration and qtwebengine that's causing this set of issues I'd hope runekit gets updated soon(tm)
In the meantime, afkwarden submenus work on my end and show their configuration options, so it could be something off on your end
I think there's some disparity between steam runescape (if you're using that) and native runescape, so there's [this fork](https://github.com/Toasterbirb/runekit) that was created to help it work (something about the display backends being different iirc)
Do any other apps such as clue solver work?

**Reply by 607ch00** [reply]
I was getting a black screen when trying to configure the map for clues. Super silly work around by enabling the fancy frame around all popup windows which lets you minimize then maximize the popup and the page redraws without being all black.  It's in my fork and building, hopefully it helps identify the real issue!

**Reply by w-hagen** [reply]
This fix was working for me until recently using the snap version of rs3 and the Appimage for Runekit. Even with this variable set now I get the same blank box as before. I'm not quite sure what changed to effect the behavior. I don't use it a lot, and with a rolling distro it can be hard to pinpoint the exact update that caused issues.

I did spend some time today to try and update RuneKit from PySide2 (Qt5) to PySide6 (Qt6)--which seemed relatively straightforward. Unfortunately while that fixed the blank/transparent screen and I can interact with the clue solver tabs it results in new error messages and it doesn't seem to be able to read the screen
> DEBUG:runekit.browser.api.RuneKitSchemeHandler:RPC: bindRegion({'x': 0, 'y': 0, 'w': 2560, 'h': 1371})
qt.core.qobject.connect: QObject::connect: Cannot queue arguments of type 'QWebEngineUrlRequestJob*'
(Make sure 'QWebEngineUrlRequestJob*' is registered using qRegisterMetaType().)

I'm not sure if this is a *new* issue or if it is related to the issues seen running with Qt5. I don't have any experience with the Qt framework though and not sure how effective I can be at debugging it. I will probably push a fork with the updated Qt6 in a few days if anyone is interested.

**Reply by hankshill** [reply]
I got the apps to show up properly using `export QTWEBENGINE_CHROMIUM_FLAGS="--no-sandbox"` but they don't seem to be able to read my screen successfully. I'm using the `runescape-launcher` package from the AUR and the RuneKit appimage. 

Here are just a few things, but basically every plugin is unusable and broken on my end: XpMeter just sits at "searching...". It will switch to 0 xp/hr if I adjust the warning time. Afk warden doesn't detect anything at all. Bank value calculator can't see my bank.

So close, but so far.

**Reply by Pen-thus** [reply]
Those apps have never worked for me to begin with - that's likely more of a problem with runekit rather than qtwebengine
For my use case I only care about vis wax, meg, and clue scrolls - all of which work for me right now, though it would be nice if afkwarden worked

**Reply by w-hagen** [reply]
My suspicion is something with RuneKit is not quite right. My digging seamed to suggest there might be some sort of pointer error with my very limited knowledge with this language. 

Qt6 is supposed to be a drop in replacement more or less for Qt5 and seeing errors further suggests something might be not quite right.  On Qt6 I was getting similar behavior, the apps would show up and I could interact with them but it was having problems reading my screen, which seems to be what is happening on Qt5 at times too.

My setup is also unkind I suspect. Wayland makes things more challenging as it is still has some issues.

**Comment by Repkap11** [comment]
@w-hagen, I also ported RuneKit from PySide2 to PySide6 to fix the glitched out QWebEngine rendering.
I ran into the same problem with "Cannot queue arguments of type 'QWebEngineUrlRequestJob*'"
I was able to fix this problem by passing "object" instead of "QWebEngineUrlRequestJob" for "successSignal" and "on_success" callback.

With this change the clue solver is partially working once again.

**Reply by pekempy** [reply]
> @w-hagen, I also ported RuneKit from PySide2 to PySide6 to fix the glitched out QWebEngine rendering. I ran into the same problem with "Cannot queue arguments of type 'QWebEngineUrlRequestJob*'" I was able to fix this problem by passing "object" instead of "QWebEngineUrlRequestJob" for "successSignal" and "on_success" callback.
> 
> With this change the clue solver is partially working once again.

You'll have more luck using [alt1-electron ](https://github.com/skillbert/alt1-electron) as this repo is unmaintained

**Comment by All-The-Foxes** [comment]
I have the same issue. A year or two ago I was able to use it fine on Ubuntu 20.04 LTS, but on Ubuntu 22.10 I have the same issue as OP. Turning off the fancy border setting just gives me solid white windows.
